### ISSUE TITLE
Extending LocalSubscription?  (updated)

### ISSUE THREAD

**Issue by chadobado** [ParsePlatform/ParseReact#16]
TLDR;

Any reason I shouldn't extend LocalSubscription to:

1) Include other client-only data
2) Have it listen for Dispatch events (ala Flux)

Thanks in advance.

......

Am trying to reconcile your library with my understanding of Flux's pattern of a single Dispatcher delegating Actions to multiple Datastores that components subscribe to.  

Simplifying this pattern down to a single interface/ObjectStore is a breath of fresh air but I have a few questions.  One of the benefits of Flux is being able to fire-and-forget actions that help marshal changes globally outside a component tree (often with data only persisted on the client and not sent to the server).  

How might this be accommodated with ParseReact?

Seems like the two primary options are to implement Flux Data Stores alongside ParseReact or extend LocalSubscriptions.
```
Local Subscriptions allow applications to subscribe to local objects, such
as the current user. React components can watch these for changes and
re-render when they are updated.
```
What are your thoughts on extending LocalSubscriptions to register a callback with a dispatcher to handle different actionTypes being dispatched through the system?  Callback interprets the action payload to handle it appropriately (encapsulating any needed logic).  From there, using ParseReact's triggerUpdate to notify subscribed components of the changes.

Seems like this approach would greatly eliminate the overhead of passing props through an entire component tree (since nested components can subscribe directly to LocalSubscription as needed) and for dealing with data between components that don't have a parent-child relationship.

Love your thoughts... thanks in advance.

**Comment by chadobado** [comment]
Generally got this working by extending LocalSubscription.  Will attempt to create a stand-alone Flux Store that uses the appropriate ParseReact methods (so is Observable) and registers a callback with the Dispatcher to update itself as needed.

Will likely push this into a [wrapper component](https://medium.com/@dan_abramov/mixins-are-dead-long-live-higher-order-components-94a0d2f9e750) to separate component's rendering concerns from [how data being consumed is fetched](https://github.com/acdlite/flummox/blob/v3.5.1/docs/docs/guides/why-flux-component-is-better-than-flux-mixin.md).  

If I'm reinventing the wheel would love to know, thanks.

**Comment by andrewimm** [comment]
So, LocalSubscription is less about client-only data, and more about client-only concepts. The current User or current Session is a pointer of sorts that only exists locally, but points to data that exists on both the server and client.

Now that there is a standard `observe()` API, you might want to explore using that to inject your data into a component. You simply need an object that takes a `subscribe(callbacks)` method, where `callbacks` is an object of the form:

```
{
  onNext: function(value) {
    // This method is called each time the subscribed value is updated
  }
}
```

The result returned from `subscribe()` should be an object with a `.dispose()` method. Each time the component updates, it will create new subscriptions from the `observe()` method, and dispose of old ones. You can find more on this proposal at [https://github.com/facebook/react/issues/3398](https://github.com/facebook/react/issues/3398).

**Comment by facebook-github-bot-4** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.
